transformer_engine/pytorch/module/layernorm_linear.py

```diff
@@ -525,6 +525,11 @@ def backward(
             # Handle custom DDP from mcore.
             if ctx.fuse_wgrad_accumulation and hasattr(weight, 'grad_added_to_main_grad'):
                 weight.grad_added_to_main_grad = True
+                wgrad = torch.empty(weight.main_grad.shape,
+                                   dtype=weight.dtype,
+                                   device=torch.cuda.current_device(),
+                                   requires_grad=False
+                                   )
```

Comment:
Could we set `wgrad` to `None`? It seems that this would avoid some memory allocation and would make it clear that the grad isn't supposed to be used.

Reply:
Setting wgrad to None, would break the M-LM usage. M-LM has a check that wgrad should not be None. (https://github.com/NVIDIA/Megatron-LM/blob/5f2877d85cb26e47ce6dcdae4b80adf376abf4e8/megatron/core/distributed/distributed_data_parallel.py#L175)
It is anyway freed after this check, this is done due to some multi-threading reason done my Torch during Bprop.

Reply:
No, we can't set `wgrad` to `None`...we are trying to prevent PyTorch from launching backward hooks in helper threads (we use backward hooks to tell us when we can launch asynchronous all-reduces / reduce-scatters).